vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridCellFocusPageIT.java

```diff
@@ -0,0 +1,117 @@
+    @Test
+    public void test_focusFirstCell() {
+        open();
+
+        getGrid().getCell(0, 0).focus();
+        assertItemResult(GridCellFocusPage.ID_ITEM_RESULT, "A");
+
+        assertItemResult(GridCellFocusPage.ID_COLUMN_RESULT,
+                GridCellFocusPage.KEY_FIRST_COLUMN);
+
+        assertItemResult(GridCellFocusPage.ID_SECTION_RESULT,
+                GridCellFocusPage.SECTION_DETAILS);
+
+
+        getGrid().getCell(1, 0).focus();
+        assertItemResult(GridCellFocusPage.ID_ITEM_RESULT, "B");
+        assertItemResult(GridCellFocusPage.ID_COLUMN_RESULT,
+                GridCellFocusPage.KEY_FIRST_COLUMN);
+        assertItemResult(GridCellFocusPage.ID_SECTION_RESULT,
+                GridCellFocusPage.SECTION_DETAILS);
+
+        getGrid().getCell(2, 1).focus();
+        assertItemResult(GridCellFocusPage.ID_ITEM_RESULT, "C");
+        assertItemResult(GridCellFocusPage.ID_COLUMN_RESULT,
+                GridCellFocusPage.KEY_SECOND_COLUMN);
+        assertItemResult(GridCellFocusPage.ID_SECTION_RESULT,
+                GridCellFocusPage.SECTION_DETAILS);
+    }
+
+    @Test
+    public void test_focusHeaderCell() {
+        open();
+
+        getGrid().getHeaderCell(0).focus();
+        assertItemResult(GridCellFocusPage.ID_ITEM_RESULT,
+                GridCellFocusPage.NO_ITEM);
+        assertItemResult(GridCellFocusPage.ID_COLUMN_RESULT,
+                GridCellFocusPage.KEY_FIRST_COLUMN);
+        assertItemResult(GridCellFocusPage.ID_SECTION_RESULT,
+                GridCellFocusPage.SECTION_HEADER);
+
+        getGrid().getHeaderCell(1).focus();
+        assertItemResult(GridCellFocusPage.ID_ITEM_RESULT,
+                GridCellFocusPage.NO_ITEM);
+        assertItemResult(GridCellFocusPage.ID_COLUMN_RESULT,
+                GridCellFocusPage.KEY_SECOND_COLUMN);
+        assertItemResult(GridCellFocusPage.ID_SECTION_RESULT,
+                GridCellFocusPage.SECTION_HEADER);
+
+    }
+
+    @Test
+    public void test_focusFooterCell() {
+        open();
+
+        getGrid().getFooterCell(0).focus();
+        assertItemResult(GridCellFocusPage.ID_ITEM_RESULT,
+                GridCellFocusPage.NO_ITEM);
+        assertItemResult(GridCellFocusPage.ID_COLUMN_RESULT,
+                GridCellFocusPage.KEY_FIRST_COLUMN);
+        assertItemResult(GridCellFocusPage.ID_SECTION_RESULT,
+                GridCellFocusPage.SECTION_FOOTER);
+
+        getGrid().getFooterCell(1).focus();
+        assertItemResult(GridCellFocusPage.ID_ITEM_RESULT,
+                GridCellFocusPage.NO_ITEM);
+        assertItemResult(GridCellFocusPage.ID_COLUMN_RESULT,
+                GridCellFocusPage.KEY_SECOND_COLUMN);
+        assertItemResult(GridCellFocusPage.ID_SECTION_RESULT,
+                GridCellFocusPage.SECTION_FOOTER);
+    }
+
+    private GridElement getGrid() {
+        return $(GridElement.class).id(GridCellFocusPage.ID_GRID);
+    }
+
+    private void assertItemResult(String resultFieldId, String expectedResult) {
```

Comment:
Would suggest a more generic name like `assertTextResult`  since this is not only used for item values, but also for other data like section and columns.

Reply:
Changed naming.